finders.go

```diff
@@ -256,8 +256,32 @@ func (q *Query) eagerAssociations(model interface{}) error {
 //
 // 	q.Where("name = ?", "mark").Exists(&User{})
 func (q *Query) Exists(model interface{}) (bool, error) {
-	i, err := q.Count(model)
-	return i != 0, err
+	tmpQuery := Q(q.Connection)
+	q.Clone(tmpQuery) //avoid mendling with original query
+
+	var res bool
+
+	err := tmpQuery.Connection.timeFunc("Exists", func() error {
+		tmpQuery.Paginator = nil
+		tmpQuery.orderClauses = clauses{}
+		tmpQuery.limitResults = 0
+		query, args := tmpQuery.ToSQL(&Model{Value: model})
+
+		// when query contains custom selected fields / executed using RawQuery,
+		// sql may already contains limit and offset
```

Comment:
Sorry, another typo.
Should be "contain", not plural.

Reply:
That's not plural, `contains` refers to `query`. That's the third person form, in present tense.